recipes/qt/all/conanfile.py

```diff
@@ -0,0 +1,676 @@
+
+            if not python_exe:
+                msg = ("Python2 must be available in PATH "
+                       "in order to build Qt WebEngine")
+                raise ConanInvalidConfiguration(msg)
+            # In any case, check its actual version for compatibility
+            from six import StringIO  # Python 2 and 3 compatible
+            mybuf = StringIO()
+            cmd_v = "{} --version".format(python_exe)
+            self.run(cmd_v, output=mybuf)
+            verstr = mybuf.getvalue().strip().split('Python ')[1]
+            if verstr.endswith('+'):
+                verstr = verstr[:-1]
+            version = tools.Version(verstr)
+            # >= 2.7.5 & < 3
+            v_min = "2.7.5"
+            v_max = "3.0.0"
+            if (version >= v_min) and (version < v_max):
+                msg = ("Found valid Python 2 required for QtWebengine:"
+                       " version={}, path={}".format(mybuf.getvalue(), python_exe))
+                self.output.success(msg)
+            else:
+                msg = ("Found Python 2 in path, but with invalid version {}"
+                       " (QtWebEngine requires >= {} & < "
+                       "{})".format(verstr, v_min, v_max))
+                raise ConanInvalidConfiguration(msg)
+
+    def config_options(self):
+        if self.settings.os != "Linux":
+            del self.options.with_icu
+        if self.settings.compiler == "apple-clang":
+            if tools.Version(self.settings.compiler.version) < "10.0":
+                raise ConanInvalidConfiguration("Old versions of apple sdk are not supported by Qt (QTBUG-76777)")
+        if self.settings.compiler == "gcc" and Version(self.settings.compiler.version) < "5.3":
+            del self.options.with_mysql
+        if self.settings.os == "Windows":
+            del self.options.with_mysql
+
+    def configure(self):
+        if self.options.opengl == "auto":
+            self.options.opengl = ("dynamic" if self.settings.os == "Windows" else "desktop")
+        if self.settings.os != 'Linux':
+            #     self.options.with_libiconv = False # QTBUG-84708
+            del self.options.with_fontconfig
+
+        if self.options.widgets and not self.options.GUI:
+            raise ConanInvalidConfiguration("using option qt:widgets without option qt:GUI is not possible. "
+                                            "You can either disable qt:widgets or enable qt:GUI")
+        if not self.options.GUI:
+            del self.options.opengl
+            del self.options.with_vulkan
+            del self.options.with_freetype
+            del self.options.with_fontconfig
+            del self.options.with_harfbuzz
+            del self.options.with_libjpeg
+            del self.options.with_libpng
+
+        if not self.options.qtmultimedia:
+            del self.options.with_libalsa
+            del self.options.with_openal
+
+        if self.settings.os != "Linux":
+            del self.options.with_libalsa
```

Comment:
Should this not go under config_options?

Reply:
you're right! I will do this later